aws/endpoints/decode.go

```diff
@@ -178,6 +179,27 @@ func custFixAppAutoscalingChina(p *partition) {
 	p.Services[serviceName] = s
 }
 
+func custFixEc2China(p *partition) {
+	if p.ID != "aws-cn" {
+		return
+	}
+
+	const serviceName = "ec2"
+	s, ok := p.Services[serviceName]
+	if !ok {
+		return
+	}
+
+	const expectHostname = `ec2.{region}.amazonaws.com`
```

Comment:
Could you provide more information about this change. Per [AWS's China endpoints](https://docs.amazonaws.cn/en_us/general/latest/gr/cnnorthwest_region.html) documentation this should be be a `ec2.cn-northwest-1.amazonaws.com.cn` endpoint.